src/metabase/channel/email.clj

```diff
@@ -122,17 +122,18 @@
 (defn- notification-recipients->emails
   [recipients payload]
   (into [] cat (for [recipient recipients
-                     :let [emails (case (:type recipient)
+                     :let [details (:details recipient)
+                           emails (case (:type recipient)
                                     :notification-recipient/user
                                     [(-> recipient :user :email)]
                                     :notification-recipient/group
                                     (->> recipient :permissions_group :members (map :email))
                                     :notification-recipient/external-email
-                                    [(-> recipient :details :email)]
+                                    [(:email details)]
                                     :notification-recipient/template
-                                    [(-> recipient :details :pattern (channel.params/substitute-params payload))]
+                                    [(not-empty (channel.params/substitute-params (:pattern details) payload (:is_optional details)))]
                                     nil)]
-                     :when (seq emails)]
+                     :when (seq (filter some? emails))]
                  emails)))
```

Comment:
I know this doesn't happen in practice yet, but if there are any `nil` emails mixed with non-nil ones, then we'll return the unfiltered list here.

How about rebinding `emails` to be filtered before doing the `when`?

Reply:
why not :D updated